### DISCUSSION TITLE
[v0.23.0]Does the ratelimit label have a priority?

### DISCUSSION THREAD

**Discussion by cddqssc** [pocketbase/pocketbase#5891] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/5a38b6c7-7f03-4b0d-affc-561eaf32fb5a)
As shown in the diagram, when I send a POST /api/request-otp request, it can first match the /api/ rule. Will it continue to match the POST /api/request-otp rule afterward?

**Comment by ganigeorgiev** [comment, marked as answer]
The resolving priority should be the following (at the moment the lookup stops on the first matching rule):

1. exact tag (e.g. `users:create`)
2. wildcard tag (e.g. `*:create`)
3. METHOD + exact path (e.g. `POST /a/b`)
4. METHOD + prefix path (e.g. `POST /a/b/`)
5. exact path (e.g. `/a/b`)
6. prefix path (e.g. `/a/b/`)

Rules from the same "type" depends on their order. I'll consider before the final v0.23 to add sorting support in the UI.

_edit: update the order of the prefix matching_

**Reply by ganigeorgiev** [reply]
On second thought, I forgot to account for the new `Audience` field _("Targeted user" in the UI)_ 
introduced with v0.23.0-rc13 since it is allowed to have 2 rules with the same label but one targeting "guest" and other "auth". Will have to think a little more on it how to adapt the rules search.

**Reply by ganigeorgiev** [reply]
I've submitted a fix in the `develop` branch that should properly accounts for the audience field and also added auto sort for the rate limit rules (_the rate limit documentation was also update to clarify the priority_).

**Reply by cddqssc** [reply]
@ganigeorgiev 
![image](https://github.com/user-attachments/assets/ab776508-41cc-4b8b-9e08-146080d5d2dc)
Thank you for your prompt response. I would like to test the latest development version and noticed that the order in the rate limit table has changed compared to before. So, can I understand "auto sort" to mean that the order from top to bottom in the table represents the actual priority?

**Reply by ganigeorgiev** [reply]
Yes, at least that was the main idea.
